### PR TITLE
[LTL] Add support for new sequence and property ops

### PR DESCRIPTION
**Pull request by dobios**
This PR adds support for the ltl ops that were recently added to the `ltl` dialect in CIRCT:  
- [x] Intersect  
- [x] Repeat
- [x] GoToRepeat
- [x] NonConsecutiveRepeat
- [x] Until  

### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Feature (or new API)

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

#### Release Notes
- [LTL] Add support for intersect, repeat, gotoRepeat, and nonConsecutiveRepeat ops.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by dobios** [comment]
The new ops should be included in the next release of firtool

File: src/main/scala/chisel3/ltl/LTL.scala [hunk 1]
```diff
@@ -159,6 +226,25 @@ object Sequence {
...
+    * times of both sequences must be identical. Equivalent to
+    * `arg0 intersect arg1 intersect ... intersect argN` in SVA.
+    */
+  def intersect(arg0: Sequence, argN: Sequence*)(implicit sourceInfo: SourceInfo): Sequence = {
+    var lhs = arg0
+    for (rhs <- argN) {
+      lhs = OpaqueSequence(LTLIntersectIntrinsic(lhs.inner, rhs.inner))
+    }
+    lhs
+  }
```
Comment (jackkoenig) [inline src/main/scala/chisel3/ltl/LTL.scala:239]:
```suggestion
  def intersect(args: Sequence*)(implicit sourceInfo: SourceInfo): Sequence = {
    require(args.nonEmpty, s"intersect requires at least one argument! Got $args")
    args.tail.foldLeft(args.head)((lhs, rhs) =>
      OpaqueSequence(LTLIntersectIntrinsic(lhs.inner, rhs.inner))
    )
  }
```
It's more canonical to write this for loop with a fold.

I get that you're sort of forcing the user to have at least one argument via the arg followed by varargs, but that has the nasty side effect of making it very annoying to call this with a `Seq[Sequence]` argument, so I think it's best just to go full varargs.

Another alternative is to overload, have this form and then have a 2nd that takes a `Seq`.

File: src/main/scala/chisel3/ltl/LTL.scala [hunk 2]
```diff
@@ -274,6 +366,25 @@ object Property {
     lhs
   }
 
+  /** Form the conjunction of two properties, where the start and end
+    * times of both sequences must be identical. Equivalent to
+    * `arg0 intersect arg1 intersect ... intersect argN` in SVA.
+    */
+  def intersect(arg0: Property, argN: Property*)(implicit sourceInfo: SourceInfo): Property = {
```
Comment (jackkoenig) [inline src/main/scala/chisel3/ltl/LTL.scala:373]: Same comment as other `intersect`

File: src/main/scala/chisel3/util/circt/LTLIntrinsics.scala [hunk 3]
```diff
@@ -77,6 +89,30 @@ private[chisel3] object LTLDelayIntrinsic {
...
 }
 
+/** A wrapper intrinsic for the CIRCT `ltl.repeat` operation. */
+private[chisel3] object LTLRepeatIntrinsic {
+
+  def apply(base: Int, more: Option[Int])(_in: Bool)(implicit sourceInfo: SourceInfo) = {
+    val params = more match {
+      case None    => Seq("base" -> IntParam(base))
+      case Some(l) => Seq("base" -> IntParam(base), "more" -> IntParam(l))
+    }
```
Comment (jackkoenig) [inline src/main/scala/chisel3/util/circt/LTLIntrinsics.scala:99]:
```suggestion
    val params = Seq("base" -> IntParam(base)) ++ more.map("more" -> IntParam(_))
    }
```
No need to repeat the `base` bit.

File: src/test/scala/chiselTests/LTLSpec.scala [hunk 4]
```diff
@@ -85,36 +85,77 @@ class LTLSpec extends AnyFlatSpec with Matchers with ChiselRunners {
...
     val s0: Sequence = a.delay()
     val s1: Sequence = s0.and(b)
     val s2: Sequence = s0.or(b)
+    val si: Sequence = s0.intersect(b)
     val s3: Sequence = s0.clock(clock)
     val p0: Property = a.eventually
     val p1: Property = p0.and(b)
     val p2: Property = p0.or(b)
+    val pi: Property = p0.intersect(b)
     val p3: Property = p0.clock(clock)
```
Comment (jackkoenig) [inline src/test/scala/chiselTests/LTLSpec.scala:132]: Please add tests for `intersect` using more than 1 argument.

**Review by jackkoenig — APPROVED** [review]
Generally LGTM but please see comments about `intersect`.

File: src/main/scala/chisel3/ltl/LTL.scala (hunk 1 above)
Comment (dobios) [reply src/main/scala/chisel3/ltl/LTL.scala:239]: oh ok, yeah that makes sense, I was just following the pattern used for `and` and `or`. Should I update those too?

Comment (jackkoenig) [reply src/main/scala/chisel3/ltl/LTL.scala:239]: In that case I think just leave it consistent (i.e. leave it as is), we can consider changing them all in a separate PR.

File: src/main/scala/chisel3/ltl/LTL.scala [hunk 5]
```diff
@@ -6,6 +6,7 @@ import chisel3._
 import chisel3.util.circt._
 import chisel3.experimental.hierarchy.{Instance, Instantiate}
 import chisel3.experimental.SourceInfo
+import firtoolresolver.shaded.coursier.ivy.PropertiesPattern.ChunkOrProperty.Prop
```
Comment (sequencer) [inline src/main/scala/chisel3/ltl/LTL.scala:9]: Why this is required?

File: src/main/scala/chisel3/util/circt/LTLIntrinsics.scala (hunk 3 above)
Comment (dobios) [reply src/main/scala/chisel3/util/circt/LTLIntrinsics.scala:99]: changed it for delay as well (I had written that part too)

File: src/main/scala/chisel3/ltl/LTL.scala (hunk 5 above)
Comment (jackkoenig) [reply src/main/scala/chisel3/ltl/LTL.scala:9]: Probably an accidental VSCode import. It loves pulling things from `firtoolresolver.shaded`...
```suggestion
```

Comment (dobios) [reply src/main/scala/chisel3/ltl/LTL.scala]: stupid metals, constantly auto-importing random stuff

**Pull request merged**